dev-docs/modules/userId.md

```diff
@@ -764,6 +764,71 @@ pbjs.setConfig({
 });
 {% endhighlight %}
 
+### PubProvided ID
+
+The PubProvided Id module allows publishers to set and pass a first party user id into the bid stream. This module has several unique characteristics:
+
+1. The module supports a user defined function, that generates an eids-style object:
+
+```
+pbjs.setConfig({
+    userSync: {
+        userIds: [{
+            name: "publisherProvided",
+            params: {
+                eidsFunction: getIdsFn   // any function that exists in the page
+            }
+        }]
+    }
+});
+```
+
+Or, the eids values can be passed directly into the `setConfig` call:
+```
+pbjs.setConfig({
+    userSync: {
+        userIds: [{
+            name: "example.com",
+            params: {
+                eids: [{
+                    source: "domain.com",
+                    uids:[{
+                      id: "value read from cookie or local storage",
+                      atype: 1,
+                      ext: {
+                          stype: "ppuid"
+                      }
+
+                  }]
+                },{
+                    source: "3rdpartyprovided.com",
+                    uids:[{
+                      id: "value read from cookie or local storage",
+                      atype: 3,
+                      ext: {
+                          stype: "sha256email"
+                      }
+                  }]
+                }]
+            }
+        }]
+    }
+});
+```
+
+In either case, bid adapters will receive the eid values after consent is validated.
+
+2. This design allows for the setting of any number of uuids in the eids object. Publishers may work with multiple ID providers and nest their own id within the same eids object.  The opportunity to link a 1st party uuid and a 3rd party generated UUID presents publishers with a unique ability to address their users in a way demand sources will understand.
+
+3. Finally, this module allows publishers to broadcast their user id, derived from in-house tech, directly to buyers within the confines of existing compliance (CCPA & GDPR) frameworks. 
+
+4. The `eids.uids.ext.stype` "source-type" extension helps downstream entities know what do with the data. Currently defined values are:
+
+- dmp - this uid comes from the in-page dmp named in eids.source
+- ppuid - this uid comes from the publisher named in eids.source
+- sha256email - this uid is based on an email hash from the entity named in eids.source
```

Comment:
@jdwieland8282 maybe this is a discussion for identity committee, but do we really want to support passing the actual hashed email through the bidstream like this? a hashed email is like a fingerprint in that you can't "reset" it - once you have it, you will always have the same link to a set of user data, even if they've asked someone upstream to reset/clear their data. since there's no efficient way (today) to tell EVERY platform in the industry to wipe data for a user, the best way today is simply to generate a new user id. this is just like apple and android allowing you to reset your MAID. identity providers that base an ID off of a hashed email are fine since they can change the ID they generate if the user has asked to reset/opt out at some point. 

my suggestion for now is to not include mentions of passing sha256email directly today, and we can discuss in committee meeting next week to see if anyone else agrees with me or if i'm just being overly precautious.  thoughts?

Reply:
Thats a fair @smenzer, I can strip that example out for now.  I would like to have this discussion in  [5775](https://github.com/prebid/Prebid.js/issues/5775) though.  I think defining what types of ids a given user id module generates is worth while.

Reply:
> Thats a fair @smenzer, I can strip that example out for now. I would like to have this discussion in [5775](https://github.com/prebid/Prebid.js/issues/5775) though. I think defining what types of ids a given user id module generates is worth while.

ok cool, i'll copy my comment above to that issue for discussion...thanks, @jdwieland8282 !